sentry_sdk/utils.py

```diff
@@ -655,11 +655,15 @@ def get_errno(exc_value):
 
 def get_error_message(exc_value):
     # type: (Optional[BaseException]) -> str
-    return (
+    value = (
         getattr(exc_value, "message", "")
         or getattr(exc_value, "detail", "")
         or safe_str(exc_value)
     )
+    notes = getattr(exc_value, "__notes__", [])
+    if notes:
+        value = "\n".join([value] + [safe_str(note) for note in notes])
```

Comment:
I am unsure whether it makes sense to add the note to the exception message; perhaps setting this as [extra data](https://docs.sentry.io/concepts/key-terms/enrich-data/) would be better

Reply:
I responded in the pull request conversation.

I think the notes add context that developers expect to be shown directly with the error message. That is what CPython does in the console.

Reply:
I don't think we need this `safe_str` stuff here, notes should anyways just be strings. This is guaranteed when using `add_note`. Although you correctly point out in your test that it is possible to add non-string values to the `__notes__` list manually, this is likely a misuse of `__notes__`, and is not something we need to support; although, we should still not crash if there is an invalid value in the `__notes__`. 

I would suggest we just filter out any non-string values from the `__notes__`.